File: packages/llm-test-harness/src/metrics/not-empty.ts
import type { MetricFn, MetricScore } from '../types.js'

/**
 * Asserts the response is not empty or whitespace-only.
 * Simple but surprisingly useful as a baseline sanity check.
 *
 * @example
 * Metrics.notEmpty()
 * Metrics.notEmpty({ minLength: 10 })
 */
export class NotEmpty implements MetricFn {
  readonly name = 'NotEmpty'
  private minLength: number

  constructor(options?: { minLength?: number }) {
    this.minLength = options?.minLength ?? 1
  }

  evaluate(text: string): MetricScore {
    const trimmed = text.trim()
    const pass = trimmed.length >= this.minLength
    return {
      name: this.name,
      pass,
      score: pass ? 1 : 0,
      reason: pass
        ? undefined
        : this.minLength === 1
          ? 'Response is empty'
          : `Response length ${trimmed.length} is below minimum ${this.minLength}`,
    }
  }
}

File: packages/llm-test-harness-python/llm_test_harness/metrics/not_empty.py
from __future__ import annotations

from ..types import MetricFn, MetricScore


class NotEmpty(MetricFn):
    """Assert the response is not empty or whitespace-only.

    Example::

        Metrics.not_empty()
        Metrics.not_empty(min_length=20)
    """

    def __init__(self, *, min_length: int = 1) -> None:
        self._min_length = min_length

    @property
    def name(self) -> str:
        return "NotEmpty"

    def evaluate(self, text: str) -> MetricScore:
        trimmed = text.strip()
        passed = len(trimmed) >= self._min_length
        if not passed:
            reason = (
                "Response is empty"
                if self._min_length == 1
                else f"Response length {len(trimmed)} is below minimum {self._min_length}"
            )
        else:
            reason = None
        return MetricScore(name=self.name, passed=passed, score=1.0 if passed else 0.0, reason=reason)
